### PR TITLE
WD-13: Unit Testing!

### PR DESCRIPTION
**Pull request by cakoshakib**
- Wrote tests for all api endpoints except `/auth` (not really sure how to make a test for that 😂)
- `go test ./tests/`
- In future, we may want to reconsider our UID design as I'm realizing it is probably not very secure (Bearer token or something of that sort prob makes more sense)

### CODE REVIEW

File: db/rds.go [hunk 1]
```diff
@@ -84,29 +94,39 @@ func SelectUserTaskByID(uid string, task_id string) Task {
 func SelectDiscordID(uid string) string {
 	query := fmt.Sprintf("SELECT discord_id FROM users WHERE uid='%s'", uid)
 	var discord_id string
-	err := db.QueryRow(query).Scan(&discord_id)
+	err := DB.QueryRow(query).Scan(&discord_id)
 	if err != nil {
-		panic(err)
+		return ""
```
Comment (markdstouffer) [inline db/rds.go:99]: What is this for? vs. "panic(err)" or just "return"

File: db/schema.go
Comment (markdstouffer) [inline db/schema.go:1]: camelCase FTW

File: tests/api_test.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+func TestGetTasks(t *testing.T) {
+	init_test()
+	// Post some data
+	var bodies []PostBody
+	bodies = append(bodies, create_task(t, "test task", 1, "2023-06-20T00:00:01Z"))
+	bodies = append(bodies, create_task(t, "random", 0, "2023-03-20T00:00:00Z"))
+	bodies = append(bodies, create_task(t, "aaa", 2, "2025-06-30T01:00:00Z"))
+
```
Comment (markdstouffer) [inline tests/api_test.go:118]: probably want to include some edge cases in here (ie the apostrophe not being allowed in content)

File: tests/api_test.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+	}
+}
+
+func TestGetTask(t *testing.T) {
+	init_test()
+	// Post some data
+	var bodies []PostBody
+	bodies = append(bodies, create_task(t, "test task", 1, "2023-06-20T00:00:01Z"))
+	bodies = append(bodies, create_task(t, "random", 0, "2023-03-20T00:00:00Z"))
+	bodies = append(bodies, create_task(t, "aaa", 2, "2025-06-30T01:00:00Z"))
```
Comment (markdstouffer) [inline tests/api_test.go:139]: is it possible to pull this code out into a separate function? reused here. in the future if we want more extensive testing it'll be better to define it in one place i think

File: tests/api_test.go
Comment (markdstouffer) [inline tests/api_test.go:1]: really cool stuff!

**Review by markdstouffer — APPROVED** [review]
LGTM! see a few small questions i left, but approved

File: db/rds.go (hunk 1 above)
Comment (cakoshakib) [reply db/rds.go:99]: return "" to match type and return instead of panic to make it so a decent error message is returned in the api rather than just an error in logs

File: tests/api_test.go (hunk 2 above)
Comment (cakoshakib) [reply tests/api_test.go:118]: rofl

File: tests/api_test.go (hunk 3 above)
Comment (cakoshakib) [reply tests/api_test.go:139]: True
